### PR TITLE
Feature/sg 76 gradient clipping

### PR DESCRIPTION
**Pull request by shairoz-deci**

### CODE REVIEW

File: src/super_gradients/training/sg_model/sg_model.py [hunk 1]
```diff
@@ -853,6 +861,10 @@ def train(self, training_params: dict = dict()):  # noqa: C901
         else:
             raise UnsupportedOptimizerFormat()
 
+        # VERIFY GRADIENT CLIPPING VALUE
+        if self.training_params.clip_grad_norm is not None and self.training_params.clip_grad_norm <= 0:
+            raise Exception('Params', 'Invalid clip_grad_norm')
```
Comment (shaydeci) [inline src/super_gradients/training/sg_model/sg_model.py:866]: Let's raise type error here instead of exception.

File: src/super_gradients/training/sg_model/sg_model.py [hunk 2]
```diff
@@ -717,6 +721,10 @@ def train(self, training_params: dict = dict()):  # noqa: C901
                 -   `sg_logger_params` : dict
 
                     SGLogger parameters
+
+                -   `clip_grad_norm` : float
```
Comment (shaydeci) [inline src/super_gradients/training/sg_model/sg_model.py:725]: Please add clip_grad_norm to super_gradients/recipes/training_hyperparams/default_train_params.yaml with a short doc like other training params.

**Review by shaydeci — CHANGES_REQUESTED** [review]

File: src/super_gradients/training/sg_model/sg_model.py (hunk 1 above)
Comment (shairoz-deci) [reply src/super_gradients/training/sg_model/sg_model.py:866]: fixed

File: src/super_gradients/training/sg_model/sg_model.py (hunk 2 above)
Comment (shairoz-deci) [reply src/super_gradients/training/sg_model/sg_model.py:725]: added

File: src/super_gradients/training/sg_model/sg_model.py (hunk 1 above)
Comment (shairoz-deci) [reply src/super_gradients/training/sg_model/sg_model.py]: fixed

**Review by shaydeci — DISMISSED** [review]
LGTM

**Review by shaydeci — APPROVED** [review]
